lib/spack/docs/mirrors.rst

```diff
@@ -266,6 +266,23 @@ is always searched first when attempting to retrieve files for an installation)
 but can be cleared with :ref:`clean <cmd-spack-clean>`; the cache directory can also
 be deleted manually without issue.
 
-Caching includes retrieved tarball archives and source control repositories, but
-only resources with an associated digest or commit ID (e.g. a revision number
-for SVN) will be cached.
+^^^^^^^^^^^^^^^^^^^^^^^
+Version Control Caching
+^^^^^^^^^^^^^^^^^^^^^^^
+
+In general, caching includes retrieved tarball archives and source control
+repositories, but only resources with an associated digest or commit ID
+(e.g. a revision number for SVN) will be cached. Some optimizations are made
+specifically for **git** repositories with the following assumptions:
+
+#. A commit hash is immutable and can be cached indefinitely.
+
+#. A tag is *assumed to be immutable*, so when spack first resolves it to
```

Comment:
Should we add something like "In practice this is not always the case, as tags can be re-created with a different history, however this case is less common." The "assumed to be" sort of implies that, but maybe we want to say it explicitly?

Reply:
I have taken your suggestion and added a link to https://git-scm.com/docs/git-tag#_on_re_tagging which clarifies this really shouldn't be done!